### PR TITLE
Require OwnerAccess to modify project member

### PR DESCRIPTION
**Pull request by BrainBalaka**
## Summary
After giving it a thought, I think we should require the user to have Owner Access instead of just Maintainer Access, for modifying the project member. 

Because it'd make less sense if a maintainer can add a member that have the Owner role. Or remove an Owner of a project.

### CODE REVIEW

**Comment by frandi** [comment]
@BrainBalaka good thinking here. But I think maintainers should be able to manage project members, as they are the people who in-charge in day to day project management. Regarding your concern, let's add validation in the code behind so that owners can be removed from a project by another owner only.

**Comment by BrainBalaka** [comment]
@frandi I've updated the code to add the validation when removing a project owner.

File: src/API/Polyrific.Catapult.Api/Controllers/ProjectMemberController.cs [hunk 1]
```diff
@@ -153,14 +156,26 @@ public async Task<IActionResult> UpdateProjectMember(int projectId, int memberId
...
-                await _projectMemberService.RemoveProjectMember(projectId, member.UserId);
-            }            
+                var member = await _projectMemberService.GetProjectMemberById(memberId);
+
+                if (member != null)
+                {
+                    int currentUserId = 0;
+
+                    if (!User.IsInRole(UserRole.Administrator))
+                        currentUserId = await _userService.GetUserId(User);
```
Comment (frandi) [inline src/API/Polyrific.Catapult.Api/Controllers/ProjectMemberController.cs]: @BrainBalaka what if we add `UserId` to the claims so we don't need to make another db connection here?

Comment (BrainBalaka) [reply src/API/Polyrific.Catapult.Api/Controllers/ProjectMemberController.cs]: Good idea. I've added the `UserId` into the claims

**Pull request merged**